ssl/ssl_locl.h

```diff
@@ -584,6 +592,13 @@ struct ssl_session_st {
         /* The ALPN protocol selected for this session */
         unsigned char *alpn_selected;
         size_t alpn_selected_len;
+        /*
+         * Maximum Fragment Length as per RFC 4366.
+         * If this value does not contain RFC 4366 allowed values (1-4) then
+         * either the Maximum Fragment Length Negotiation failed or was not
+         * performed at all.
+         */
+        uint8_t max_fragment_length;
```

Comment:
I wonder if something like "max_fragment_len_mode" is better (here and in ctx/session etc)? I got myself confused at one point reading the code thinking that this was a "real" length.